File: 프로그래머스/unrated/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<int> delete_list) {
    vector<int> answer;
    for(int a : arr){
        bool isCheck = true;
        for (int d : delete_list){
            if(a == d){
                isCheck = false;
                break;
            }
        }
        if(isCheck) answer.push_back(a);
    }
    return answer;
}

File: 프로그래머스/unrated/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[] delete_list) {
        List<Integer> answer = new ArrayList<Integer>();
        for(int a : arr){
            boolean isCheck = true;
            for(int d : delete_list){
                if( a== d){
                    isCheck = false;
                    break;
                }
            }
            if(isCheck){
                answer.add(a);
            }
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}